### DISCUSSION TITLE
git-submodule current beta state

### DISCUSSION THREAD

**Discussion by codezninja** [renovatebot/renovate#13789] [Request Help]
Hi. I'm a little confused about this beta feature. From what I can tell from all the different PRs and discussions about this starting with https://github.com/renovatebot/renovate/issues/10149. 

If I've set up the unprotected ssh key inside the container where renovate is running and the `.gitmodules` is using ssh I should be able to somewhat use this feature. But I'm getting the following authentication error when renovate is trying to access private git submodule.

```
 WARN: Error mapping git submodules during extraction (repository=MyOrg/MyRepo)
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "--symref",
             "MyOrg/.Repo",
             "HEAD"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "fatal: could not read Username for 'ghe.company.com': No such device or address\n",
         "stack": "Error: fatal: could not read Username for 'ghe.company.com': No such device or address\n\n    at Object.action (/usr/src/app/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/usr/src/app/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
       }
```

I know the ssh key works cause I am using it to update private terraform modules from the same ghe url but I'm guessing I'm missing something config-wise on this beta feature

**Comment by rarkins** [comment]
Private git submodules are noted as unsupported in the git-submodules documentation

**Reply by codezninja** [reply]
yeah I've seen that but I was curious on the comments I saw in the discussion issue. https://github.com/renovatebot/renovate/issues/10149#issuecomment-850955026. There were other Issues as well that made the same point so figured it was somehow working but not widely supported.

**Reply by viceice** [reply]
It's official not supported, those other solutions are workarounds found by users.

**Reply by chgeo** [reply]
What about the usual authentication practises like git's `credential.helper` or `.netrc` files (see [here](https://stackoverflow.com/a/63939958))?

It's maybe not the cleanest thing to rely on global configuration, but in isolated environments w/o external code running that's often good enough.  We use it in our CI systems to clone private submodules.

**Reply by codezninja** [reply]
@chgeo I couldn't get it working using those methods but there are other users that have successfully done it.
